### ISSUE TITLE
How to create the new Points object? 

### ISSUE THREAD

**Issue by alexandrefierro** [metno/titanlib#19]
Greetings:

Could you please provide a concise example on how to create this new "Points" object in version 0.3.+ as I was unable to find any documentation nor examples on GITHUB. In ancillary versions of Titanlib, users had to specify three separate arrays lats, lons and elevs, which I found clearer. These have now been merged into a Points object [list, array?] but no description is provided to users on how to build such an object in Python and what its size/dimensions should be.

Cheers and many thanks for your help,

**Comment by alexandrefierro** [comment]
Hello, Cristian:
I finally got the buddy check function to work thanks to the examples; Coming from the Fortran realm, I still have difficulties wrapping my head around the usage of lists in data computations.  It would make, in my humble view, all number crunching applications/operations more intuitive if numpy arrays were utilized as the convention in Python (in addition to all arrays being defined explicitly before the executable code begins as done in Fortran for clarity/code transparency).
Best,

**Comment by cristianlussana** [comment]
Hi again,
I am glad the examples help. I miss Fortran sometimes too. 
Best, Cristian

**Comment by tnipen** [comment]
I can add some comments on the reasons behind going from separate arrays of lat,lon,elev to a "Points" object that encapsulates all metadata about the observation points. First of all, I totally agree that using latitude, longitude, and elevation arrays is more understandable and easier to use, but in the end we switched it for these reasons:

1) The main reason is to avoid having to change the function signatures in titanlib, if we in the future need to add new metadata fields. For example, we recently added the possibility to specify points in either Cartesian coordinates (i.e. on a projected grid) or Geodetic coordinates (i.e. lat/lon). Without the points object, a new parameter would have to be added to every titanlib function that specifies what the coordinates of lat and lon are. This would then break all existing code that uses the library. Instead, only the Points object had to be modified to allow for this feature.
2) A second (but much more minor) reason is that with a Points object, we only need to create the binary search tree of positions once, and can cache the results internally in the object such that subsequent calls can reuse the search tree. This is probably not an issue at all for smaller observation datasets, but perhaps could be an issue for bigger datasets.

I've added an example of how to create the Points object on the wiki page. Thanks for the feedback!
